### ISSUE TITLE
Fatal error renderPageNavMenuItem()

### ISSUE THREAD

**Issue by al3xg0r** [robiso/wondercms#269]
Hi. After installing and creating a menu item, errors popped up:

```
Fatal error: Uncaught TypeError: Argument 1 passed to Wcms::renderPageNavMenuItem() must be an object, null given, called in F:\OpenServer\domains\test.local\index.php on line 1732 and defined in F:\OpenServer\domains\test.local\index.php:2363 Stack trace: 
#0 F:\OpenServer\domains\test.local\index.php(1732): Wcms->renderPageNavMenuItem() 
#1 F:\OpenServer\domains\test.local\themes\Sky\theme.php(47): Wcms->menu() 
#2 F:\OpenServer\domains\test.local\index.php(1618): require_once('F:\\OpenServer\\d...') 
#3 F:\OpenServer\domains\test.local\index.php(177): Wcms->loadThemeAndFunctions() 
#4 F:\OpenServer\domains\test.local\index.php(16): Wcms->render() 
#5 {main} thrown in F:\OpenServer\domains\test.local\index.php on line 2363
```

php - 7.4
wcms - 3.3.2

**Comment by robiso** [comment]
Hi @al3xg0r, after debugging this with @slavenstancic, we're going to require more data to debug this:

1. What server are you running on? (Apache, NGINX, Windows IIS, Caddy)
2. Could you please also provide your database.js file, please make sure to remove the "password" and "loginURL".

We're assuming your menu must be somehow empty, although we're unsure at this point.

**Comment by al3xg0r** [comment]
> 

1. Apache 2.4 + NGINX 1.17 (Open Server)
2.
`
{
    "config": {
        "siteTitle": "Website title",
        "theme": "Sky",
        "defaultPage": "home",
        "login": "",
        "forceLogout": false,
        "forceHttps": false,
        "saveChangesPopup": false,
        "password": "",
        "lastLogins": {
            "2022\/07\/20 14:26:49": "127.0.0.1"
        },
        "lastModulesSync": "2022\/07\/20",
        "customModules": {
            "themes": {},
            "plugins": {}
        },
        "menuItems": {
            "0": {
                "name": "Home",
                "slug": "home",
                "visibility": "show",
                "subpages": {}
            },
            "1": null,
            "": {
                "name": "About",
                "slug": "about",
                "visibility": "show",
                "subpages": {}
            }
        }
    },
    "pages": {
        "404": {
            "title": "404",
            "keywords": "404",
            "description": "404",
            "content": "<center><h1>404 - Page not found<\/h1><\/center>",
            "subpages": {}
        },
        "home": {
            "title": "Home",
            "keywords": "Enter, page, keywords, for, search, engines",
            "description": "A page description is also good for search engines.",
            "content": "<pre>Welcome to your website<\/pre>\n\n<blockquote><p>Your password for editing everything is: <b>TmjOMina<\/b><\/p><\/blockquote>\n\n<p><a href=\"http:\/\/tondocs.local\\\/loginURL\" class=\"button\">Click here to login<\/a><\/p>\n\n<p>To install an awesome editor, open Settings\/Plugins and click Install Summernote.<\/p>",
            "subpages": {}
        },
        "blog": {
            "title": "How to",
            "keywords": "Enter, keywords, for, this page",
            "description": "A page description is also good for search engines.",
            "content": "<h2>Easy editing<\/h2>\n<p>After logging in, click anywhere to edit and click outside to save. Changes are live and shown immediately.<\/p>\n\n<h2>Create new page<\/h2>\n<p>Pages can be created in the Settings.<\/p>\n\n<h2>Start a blog or change your theme<\/h2>\n<p>To install, update or remove themes\/plugins, visit the Settings.<\/p>\n\n<h2><b>Support WonderCMS<\/b><\/h2>\n<p>WonderCMS is free for over 12 years.<br>\n<a href=\"https:\/\/swag.wondercms.com\" target=\"_blank\"><u>Click here to support us by getting a T-shirt<\/u><\/a> or <a href=\"https:\/\/www.wondercms.com\/donate\" target=\"_blank\"><u>with a donation<\/u><\/a>.<\/p>",
            "subpages": {}
        }
    },
    "blocks": {
        "subside": {
            "content": "<h2>About your website<\/h2>\n\n<br>\n<p>Website description, contact form, mini map or anything else.<\/p>\n<p>This editable area is visible on all pages.<\/p>"
        },
        "footer": {
            "content": "&copy;2022 Your website"
        }
    }
}
`

**Comment by robiso** [comment]
@al3xg0r it seems like your menu was somehow broken:

```
        "menuItems": {
            "0": {
                "name": "Home",
                "slug": "home",
                "visibility": "show",
                "subpages": {}
            },
            "1": {
                "name": "About",
                "slug": "about",
                "visibility": "show",
                "subpages": {}
            }
        }
```

The above should fix it. Do you possibly have any steps to reproduce the issue that broke your menu?

**Comment by al3xg0r** [comment]
> 

The fix helped, but I figured out why it appears, when you press "Move down" in the "Menu" settings this error pops up again

**Comment by robiso** [comment]
@al3xg0r the issue has been identified and fixed bt @slavenstancic in the upcoming WonderCMS version.

Regarding the double slash in the URL, this seems like something to do with the NGINX config. Would you mind sharing it (without any sensitive info you do not wish to disclose).

**Comment by al3xg0r** [comment]
already figured it out
I'm waiting for a fix
thanks

**Comment by robiso** [comment]
Reordering first item now published on latest WonderCMS 3.3.3 version.
Feel free to login to your WonderCMS website, open settings and "Check for updates".

The update should then show up. 🚀
